q2_types/feature_data/_format.py

```diff
@@ -100,34 +100,53 @@ class TSVTaxonomyFormat(model.TextFileFormat):
     """
     HEADER = ['Feature ID', 'Taxon']
 
-    def sniff(self):
+    def _check_n_records(self, n=None):
         with self.open() as fh:
             data_lines = 0
             header = None
-            while data_lines < 10:
-                line = fh.readline()
+
+            file_ = enumerate(fh) if n is None else zip(range(n), fh)
+
+            for i, line in file_:
+                i = i + 1
```

Comment:
I think I know why you are doing this, but for the sake of future readers of the code, can you explain?

Reply:
I can definitely add a comment to describe line 111.

Reply:
Thanks for adding that comment! I don't think that is why you are offsetting `i`, though --- aren't you doing it so that you can print it in the error messages so that it reflects the logical line number in the file? Speaking of, looks like [L134](https://github.com/qiime2/q2-types/pull/221/files#diff-c7c5016b403f1d6d2fb45764b58ae9d6R134) has a reference to the line number, but, it isn't actually used in the error message.